core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -288,6 +288,7 @@ public synchronized IndexShard createShard(int sShardId, ShardRouting routing) {
             indicesLifecycle.afterIndexShardCreated(indexShard);
             settingsService.addListener(indexShard);
             shards = newMapBuilder(shards).put(shardId.id(), indexShard).immutableMap();
+            indexServicesProvider.getIndexingMemoryController().forceCheck();
```

Comment:
> Actually, why not simply start up as inactive, init the buffers to the INACTIVE values, and then the first indexing op that arrives will make it active like normal?

That was my original thought as well, but then I remembered you were very worried about indexing into a small buffer. When one creates an index by indexing into it that index will likely get many indexing requests at once, the first will trigger a forced check and the others will fly through into the engine. Figured we better not have to worry about it but If you're good with that, I'm good :)